pkg/controller/apigateway/restapi/setup.go

```diff
@@ -0,0 +1,273 @@
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithInitializers(),
+			managed.WithPollInterval(o.PollInterval),
+			managed.WithLogger(o.Logger.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type custom struct {
+	Client apigwclient.Client
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.RestAPI, obj *svcsdk.GetRestApiInput) error {
+	obj.RestApiId = aws.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.RestAPI, obj *svcsdk.RestApi, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	cr.SetConditions(xpv1.Available())
+	return obs, nil
+}
+
+func postCreate(_ context.Context, cr *svcapitypes.RestAPI, resp *svcsdk.RestApi, cre managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+
+	meta.SetExternalName(cr, aws.StringValue(resp.Id))
+	return cre, nil
+}
+
+func (c *custom) preUpdate(ctx context.Context, cr *svcapitypes.RestAPI, obj *svcsdk.UpdateRestApiInput) error {
+	cur := &svcapitypes.RestAPIParameters{
+		Region: cr.Spec.ForProvider.Region,
+	}
+
+	rapi, err := c.Client.GetRestAPIByID(ctx, aws.String(meta.GetExternalName(cr)))
+	if err != nil {
+		return errors.Wrap(err, "cant get rest api")
+	}
+
+	cur.Name = rapi.Name
+	rapi.Policy = unescapePolicy(rapi.Policy)
+
+	if err := lateInitialize(cur, rapi); err != nil {
+		return errors.Wrap(err, "cant late init current restApi")
+	}
+
+	if cr.Spec.ForProvider.Policy != nil {
+		pol, err := normalizePolicy(cr.Spec.ForProvider.Policy)
+		if err != nil {
+			return errors.Wrap(err, "cant normalize managed resource policy")
+		}
+
+		cr.Spec.ForProvider.Policy = pol
+	}
+
+	pOps, err := apigwclient.GetPatchOperations(&cur, &cr.Spec.ForProvider)
+	if err != nil {
+		return errors.Wrap(err, "cant compute patch preUpdate")
+	}
+
+	obj.PatchOperations = pOps
+	obj.RestApiId = aws.String(meta.GetExternalName(cr))
+
+	return nil
+}
+
+func policiesAreKindOfTheSame(a *string, b *string) (bool, error) {
+	if a != nil && b != nil {
+		aPol, bPol, err := parsePolicies(a, b)
+		if err != nil {
+			return false, errors.Wrap(err, "cant parse policies")
+		}
+		polPatch, err := apigwclient.GetJSONPatch(aPol, bPol)
+		if err != nil {
+			return false, errors.Wrap(err, "cant compute jsonpatch")
+		}
+
+		for _, p := range polPatch {
+			re := regexp.MustCompile(`/Statement/(\d+)/Resource`)
+			if p.Operation != "replace" || !re.MatchString(p.Path) {
+				return false, nil
+			}
+
+			index, _ := strconv.Atoi(re.FindString(p.Path))
+			fmt.Println(aPol["Statement"].([]interface{})[index].(map[string]interface{})["Resource"])
+			if aPol["Statement"].([]interface{})[index].(map[string]interface{})["Resource"] != "execute-api:/*/*/*" {
```

Comment:
Looks like a fmt.Println left in the code

Reply:
will remove 👍